### PR TITLE
k0sctl: init at 0.9.0

### PR DESCRIPTION
**Pull request by NickCao**
###### Motivation for this change

add k0sctl

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/k0sctl/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    sha256 = "sha256-aW7x2XfeFU0z3lwPTsDHudHjdwTtfASgrbKGddVb6Rs=";
+  };
+
+  vendorSha256 = "sha256-bsXXWyeZXZLV6igEvyvPpS92FruGiLDx/5CCTKPe0EU=";
+
+  meta = {
+    description = "A bootstrapping and management tool for k0s clusters.";
+    homepage = "https://k0sproject.io/";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ nickcao ];
```
Comment (musfay) [inline pkgs/applications/networking/cluster/k0sctl/default.nix:23]:
```suggestion
  meta = with lib; {
    description = "A bootstrapping and management tool for k0s clusters.";
    homepage = "https://k0sproject.io/";
    license = licenses.asl20;
    maintainers = with maintainers; [ nickcao ];
```

File: pkgs/applications/networking/cluster/k0sctl/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "k0sctl";
+  version = "0.9.0";
+
+  src = fetchFromGitHub {
+    owner = "k0sproject";
+    repo = "k0sctl";
```
Comment (musfay) [inline pkgs/applications/networking/cluster/k0sctl/default.nix:12]:
```suggestion
    repo = pname;
```

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -7530,6 +7530,12 @@
     githubId = 1665818;
     name = "Philipp Eder";
   };
+  nickcao = {
+    name = "Nick Cao";
+    email = "nickcao@nichi.co";
+    github = "NickCao";
+    githubId = 15247171;
+  };
```
Comment (musfay) [inline maintainers/maintainer-list.nix:7538]: This should be in a seperate commit with the message `maintainers: add <name>`. Make sure that the commit comes before the package.

**Review by musfay — CHANGES_REQUESTED** [review]

**Review by musfay — APPROVED** [review]
Thank you, lgtm

**Comment by roberth** [comment]
Thanks @NickCao and @musfay!

**Pull request merged**